Sources/SwiftDiscord/Channel/DiscordMessage.swift

```diff
@@ -269,11 +283,19 @@ public struct DiscordEmbed : JSONAble {
             - parameter iconUrl: The iconUrl for this author's icon.
             - parameter url: The url for this author.
         */
-        public init(name: String?, iconUrl: URL?, url: URL?) {
+        public init(name: String, iconUrl: URL? = nil, url: URL? = nil) {
             self.name = name
             self.iconUrl = iconUrl
             self.url = url
-            self.proxyUrl = nil
+            self.proxyIconUrl = nil
+        }
+
+        /// For testing
+        internal init(name: String, iconURL: URL?, url: URL?, proxyURL: URL?) {
```

Comment:
I don't think internal is really needed here? iirc you only need to specify a lower visibility if it's a protocol extension.